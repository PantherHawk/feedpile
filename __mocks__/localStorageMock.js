const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"test","identities":[{"user_id":123}]}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
