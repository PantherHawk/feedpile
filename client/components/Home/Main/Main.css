.main-container {
  padding-top: 85px;
  width: 100%;
  text-align: center;
  background-color: #f2f2f2;
  color: #333;
  padding-bottom: 100px;
}

.tagline {
  text-align: center;
  font-size: 24px;
}

.image-title-container {
  margin-top: 75px;
}

.title {
  margin-top: 0;
  font-family: 'Bitter', serif;
  font-size: 60px;
  margin-bottom: 20px;
}

.main-image {
  padding-bottom: 20px;
}

.start-btn {
  margin-top: 30px;
}
