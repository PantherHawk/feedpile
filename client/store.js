import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const composeSetup = process.env.NODE_ENV !== 'production' && typeof window === 'object' &&
window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose;

const store = createStore(rootReducer, composeSetup (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
