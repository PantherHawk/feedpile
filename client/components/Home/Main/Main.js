import React from 'react';
import Card from 'material-ui/Card'
import RaisedButton from 'material-ui/RaisedButton'
import './Main.css'

const Main = (props) => {
  return (
    <div className="main-container" style={{backgroundColor: "#304FFE"}}>
    <Card className="main-card" zDepth={0} style={{backgroundColor: "#304FFE"}}>
      <div className="main" >
        <i className="feedpile-logo"></i>
        <a href="#" className="brand-logo center"><img className="main-image" src="https://s13.postimg.org/8y3id7gjr/book-logo-main.png"/></a>
        <ul>
          <li><h1 className="main-h1">Feedpile</h1></li>
          <br/>

          <li><RaisedButton
          label="Get Started"
          primary={true}
          className="sign-up"/></li>
        </ul>
      </div>
      </Card>
    </div>
  )
}

export default Main;
