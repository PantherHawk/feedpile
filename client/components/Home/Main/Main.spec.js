import React from 'react';
import Main from './Main'
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import '../../../../__mocks__/localStorageMock'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {Provider} from 'react-redux';
import store from '../../../store'

test('Landingpage: Main section snapshot test', () => {

  const component = shallow(
    <Main />
    );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Landingpage: Main section should render a background, tagline, and image', () => {
  const component = render(<MuiThemeProvider><Provider store={store}><Main /></Provider></MuiThemeProvider>);
  expect(component.find('.main-container').length).toEqual(1);
  expect(component.find('.tagline').length).toEqual(1);
  expect(component.find('.main-image').length).toEqual(1);
})
