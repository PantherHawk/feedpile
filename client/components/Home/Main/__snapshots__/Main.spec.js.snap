// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landingpage: Main section snapshot test 1`] = `
<div
  className="main-container"
  style={
    Object {
      "backgroundColor": "#00BCD4",
    }
  }
>
  <div
    className="tagline-container"
  >
    <h1
      className="tagline"
    >
      Simply a refreshing way to read the news
    </h1>
  </div>
  <div
    className="main"
  >
    <img
      className="main-image"
      src="https://s13.postimg.org/8y3id7gjr/book-logo-main.png"
    />
  </div>
  <h1
    className="main-h1"
  >
    Feedpile
  </h1>
</div>
`;
